org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -104,4 +124,93 @@ protected void initializeTypeProvider(XtextResourceSet resourceSet) {
 		resourceSet.setClasspathURIContext(getJavaProject(resourceSet));
 	}
 
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, CharSequence content) {
+		return createDslFile(fileName, fileExtensionProvider.getPrimaryFileExtension(), content);
+	}
+
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param fileExtension
+	 * 		The extension of the file.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, String fileExtension, CharSequence content) {
+		try {
+			return IResourcesSetupUtil.createFile(javaProject.getElementName(), fileName, fileExtension, content.toString());
+		} catch (InvocationTargetException | CoreException | InterruptedException e) {
+			throw new RuntimeException(e);
+		}
+	}
+	
+	/**
+	 * Verifies whether the content assistant provides the expected proposals on the given cursor position.
+	 * 
+	 * @param text
+	 * 		The editor's input text. The text must contain the {@link #c} special symbol indicating the current cursor position.
+	 * 
+	 * @param expectedProposals
+	 * 		The list of proposals that are expected to be offered whenever the content assistant is triggered on the given cursor position.
+	 * 
+	 * @since 2.35
+	 */
+	protected void assertContentAssistant(CharSequence text, List<String> expectedProposals) {
```

Comment:
Since we're creating a new API, could we take the chance to allow for expected proposals to be specified also as a string where proposals are separated by a new line? This would allow for easier TDD or test adjustments: it's easier to specify or update a string than to create a list of proposals or update it. Turning the string into a list or array of strings is easy with Java streams. @szarnekow wdyt?